src/Microsoft.Azure.ServiceBus.KeyVault/KeyVaultPlugin.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.ServiceBus.KeyVault
+{
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.ServiceBus.Core;
+    using System.Security.Cryptography;
+    using System.IO;
+    using System.Text;
+
+    /// <summary>
+    /// Provides Azure KeyVault functionality for Azure Service Bus.
+    /// </summary>
+    public class KeyVaultPlugin : ServiceBusPlugin
+    {
+        private readonly string secretName;
+        private readonly string keyVaultEndpoint;
+        private KeyVaultSecretManager secretManager;
+
+        /// <summary>
+        /// Gets the name that is used to identify this plugin.
+        /// </summary>
+        public override string Name => "KeyVaultPlugin";
+
+        /// <summary>
+        /// Creates a new instance of an <see cref="KeyVaultPlugin"/>.
+        /// </summary>
+        /// <param name="encryptionSecretName">The name of the secret used to encrypt / decrypt messages.</param>
+        /// <param name="options">The <see cref="KeyVaultPluginOptions"/> used to create a new instance.</param>
+        public KeyVaultPlugin(string encryptionSecretName, KeyVaultPluginOptions options)
+        {
+            if (string.IsNullOrEmpty(encryptionSecretName))
+            {
+                throw new ArgumentNullException(nameof(encryptionSecretName));
+            }
+
+            if (options != null)
+            {
+                if (string.IsNullOrEmpty(options.KeyVaultClientId))
```

Comment:
This doesn't belong here imo.
Make `KeyVaultPluginOptions` constructor the only option to build a proper object and then this code doesn't have to worry for individual properties. I.e. if you passed `KeyVaultPluginOptions`'s ctor, then all properties were validated.